jvm_version.go

```diff
@@ -0,0 +1,56 @@
+package libjvm
+
+import (
+	"strings"
+
+	"github.com/heroku/color"
+	"github.com/paketo-buildpacks/libpak"
+	"github.com/paketo-buildpacks/libpak/bard"
+)
+
+type JVMVersion struct {
+	Logger bard.Logger
+}
+
+func (jvmVersion JVMVersion) GetJVMVersion(appPath string, cr libpak.ConfigurationResolver) (string, error) {
+	version, explicit := cr.Resolve("BP_JVM_VERSION")
+
+	if !explicit {
+		manifest, err := NewManifest(appPath)
+		if err != nil {
+			return version, err
+		}
+
+		javaVersion := ""
+
+		buildJdkSpecVersion, ok := manifest.Get("Build-Jdk-Spec")
+		if ok {
+			javaVersion = buildJdkSpecVersion
+		}
+
+		buildJdkVersion, ok := manifest.Get("Build-Jdk")
+		if ok {
+			javaVersion = buildJdkVersion
+		}
+
+		if len(javaVersion) > 0 {
+			javaVersionFromMaven := extractMajorVersion(javaVersion)
+			f := color.New(color.Faint)
+			jvmVersion.Logger.Header(f.Sprint("Context specific overrides:"))
+			jvmVersion.Logger.Body(f.Sprintf("$BP_JVM_VERSION \t\t %s \t\tthe Java version, extracted from main class", javaVersionFromMaven))
```

Comment:
```suggestion
			jvmVersion.Logger.Body(f.Sprintf("Using Java version %s extracted from METADATA.MF", javaVersionFromMaven))
```

I don't think we need to log all of this. The version selected will be visible when the download happens. 

For example:

```
  BellSoft Liberica JRE 11.0.13: Contributing to layer
    Downloading from https://github.com/bell-sw/Liberica/releases/download/11.0.13+8/bellsoft-jre11.0.13+8-linux-amd64.tar.gz
```

I think the key bit is to let the user know where we sourced the version from so they know why it's not the default. What do you think of the suggestion above?

The other option would be to move the logging statement up to the if block where it pulls the property out from the manifest.mf, that way we could tell them exactly which property selected it. I'm not sure we need to go that far though, there should be only one in a manifest so it should be obvious which is being used. We just need to clue the user into where we found that.

Reply:
good point, we've updated the message + rebased the PR